go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/square/go-jose/v3
 go 1.12
 
 require (
+	github.com/btcsuite/btcd v0.20.1-beta
```

Comment:
Is there another package implementing `secp256k1` we could use here? I'm a bit wary of having this library depend on the entirety of btcd, it'd be nice if there was just a package for the curve split out somewhere.

Reply:
What about https://github.com/ethereum/go-ethereum/tree/master/crypto/secp256k1? I used this to extend x509 to support der/pem encoding for secp256k1.

Reply:
I think that'd be a better package to use. Requires CGO, so just need to make sure we use build tags to ensure we can still compile with CGO disabled, but I think this is much better than the btcd package.

Reply:
Though also again like the btcd version we'd want to pull in only the secp256k1 implementation as a dep, not the whole of go-ethereum.

Reply:
I agree. You'd need only the secp256k1 part, not the whole go-ethereum lib.

Reply:
Since this is (at least) a 2nd case where the secp256k1 is needed separately, we could put it in a separate git -- it would require to follow the new updates on go-ethereum.

Reply:
@csstaub @alenhorvat https://github.com/decred/dcrd has only crypto stuff, not the whole btcd project, this PR has been updated it.

It's also written 100% in Go, so no C dependencies needed to build it. This is the main reason why we chose this package in this PR. We don't want any external build dependency to ease the building process of go-jose and our projects.

Reply:
or a more proper one would be: https://github.com/btcsuite/btcd